packages/developer-console-spa/src/components/ConfigureDatabase.tsx

```diff
@@ -1,18 +1,234 @@
-import { Card, CardBody, Title } from '@patternfly/react-core';
-import UnderDevelopment from './UnderDevelopment';
+import {
+  Card,
+  CardBody,
+  Stack,
+  StackItem,
+  Button,
+  FlexItem,
+  Flex,
+  SearchInput,
+  Gallery,
+  GalleryItem,
+  CardHeader,
+  CardActions,
+  CardTitle,
+  CardFooter,
+  Modal,
+  DropdownItem,
+  Dropdown,
+  KebabToggle,
+  EmptyState,
+  EmptyStateIcon,
+  EmptyStateBody,
+} from '@patternfly/react-core';
+import { useContext, useState } from 'react';
+import { AppContext } from '../context/AppContext';
+import CreateDBForm from './database/CreateDBForm';
+import Header from './Header';
+import Loader from './Loader';
+import { ReactComponent as StorageIcon } from '../assets/storage_black_24dp.svg';
+import {
+  Link,
+  Route,
+  Switch,
+  useRouteMatch,
+} from 'react-router-dom';
+import UpdateCouchDBConfig from './database/UpdateDBConfig';
 
-function ConfigureDatabase ( props: any ) {
-  return (
-    <>
-      <Card isRounded>
-        <CardBody>
-          <Title headingLevel="h1">Configure Database</Title>
+function ConfigureCouchDB() {
+  const { app, loading: appLoading } = useContext(AppContext);
+  const [isCreateDBFormOpen, setIsCreateDBFormOpen] = useState(false);
+  const [expandedMenuCardId, setExpandedMenuCardId] = useState('');
+  const [searchTerm, setSearchTerm] = useState('');
+
+  let { url } = useRouteMatch();
+  const onCardMenuSelect = () => {
+    setExpandedMenuCardId('');
+  };
+  const onCardMenuToggle = (id: string) => {
+    setExpandedMenuCardId(id === expandedMenuCardId ? '' : id);
+  };
+
+  const dropdownItems = ( db: any ) => {
+    if ( url.endsWith('/') ) {
+      url = url.substring(0, url.length - 1);
+    }
+    return [
+      <Link to={`${url}/${db.name}`} key={db.name}>
+        <DropdownItem component="span">
+          Configure
+        </DropdownItem>
+      </Link>
+    ];
+  };
+
+  /**
+   * Escapes the string for special characters
+   */
+  const escapeString = (string: string) => {
+    return string.replace(/[-\\^$*+?.()|[\]{}]/g, '\\$&');
+  };
+
+  const filteredDBs = app.database?.databases.filter((db) =>
+    db.name.includes(escapeString(searchTerm.trim()))
+  );
+
+  const openCreateDBForm = () => {
+    setIsCreateDBFormOpen(true);
+  };
+
+  const handleModalClose = () => {
+    setIsCreateDBFormOpen(false);
+  };
+
+  /**
+   *
+   * @param name This function will be uncommented once we have Fauxton UI is available
+
+  const openFauxtonGUI = (name: string) => {
+    window.open(`${process.env.REACT_APP_FAUXTON_URL}/${name}/_all_docs`, '_blank' );
+  }*/
+
+  const getEmptyState = () => {
```

Comment:
This one too. Wrap it inside a useCallback, as its a function call inside render